test/jsonapi.js

```diff
@@ -311,7 +320,7 @@ describe('Creation of new resources', () => {
   it('should add reverse relationship when inserting new resource', () => {
     const updatedState = reducer(state, apiCreated(taskWithTransaction));
 
-    const { data: taskRelationship } = updatedState.transactions.data[0].relationships.task;
+    const taskRelationship = updatedState.transactions.data[0].relationships.task;
 
     expect(taskRelationship.type).toEqual(taskWithTransaction.type);
     expect(taskRelationship.id).toEqual(taskWithTransaction.id);
```

Comment:
Ah yes you are right there! I should have updated the check on taskWithTransaction a couple of lines down. Have just pushed a fix.

I understand the nerves around changes in the tests. Unfortunately the tests used mock data and it was the data I needed to change.

Both `updateEntity()` and `createEntity()` would only pass the `json.data` back from the API and we also need `json.included` so I had to pass the whole `json` variable through as on `readEndpoint()`. This meant the data structure for the tests had to change to mirror the new data structure. This join is not currently tested, maybe we should add some?